lib/core/storages/memory.ts

```diff
@@ -1,21 +1,20 @@
-import { IStorage } from "../storage";
+import { IPromiseStorage, IScheduleStorage } from "../storage";
 import { DurablePromise } from "../promise";
+import { Schedule } from "../schedule";
 
-export class MemoryStorage implements IStorage {
+export class MemoryPromiseStorage implements IPromiseStorage {
   private promises: Record<string, DurablePromise> = {};
-
   constructor() {}
 
-  async rmw<P extends DurablePromise | undefined>(
-    id: string,
-    f: (promise: DurablePromise | undefined) => P,
-  ): Promise<P> {
-    const promise = f(this.promises[id]);
-    if (promise) {
-      this.promises[id] = promise;
+  async rmw<P extends DurablePromise | undefined>(id: string, f: (item: DurablePromise | undefined) => P): Promise<P> {
+    const item = f(this.promises[id]);
+    if (item) {
```

Comment:
We can revert this back to 
```ts
if (promise) {
  this.promises[id] = item;
}

The if statement acts as a typeguard and since the type is `DurablePromise | undefined`, if it's not undefined then it must be a `DurablePromise`.